Azure/azure-cli
# [AppConfig] BREAKING CHANGE: az appconfig feature filter add: Support adding JSON objects as feature filter parameter values

**Description**
Fix #16374 

- **Affected Command:**
`Az appconfig feature filter add -n TestAppConfig --feature Beta –filter-name TestFilter --filter-parameters name1=value1 name2=value2`

- **Current Behavior:**
Always treat the `value` in `--filter-parameters name=value` as string value.

- **New Behavior:**
Accept only _escaped JSON string_ as `value` in `--filter-parameters name=value`, and convert it to the correct data type accordingly.


**Testing Guide**

The following command should fail since the value is not a JSON escaped string:
`az appconfig feature filter add -n TestAppConfig --feature Beta --filter-name TestFilter --filter-parameters stringValue=SomeText`

This modified version should pass:
`az appconfig feature filter add -n TestAppConfig --feature Beta --filter-name TestFilter --filter-parameters stringValue=\"SomeText\"`



**History Notes**


[AppConfig] BREAKING CHANGE: az appconfig feature filter add:
- Support adding JSON objects as feature filter parameter values
- Accept only JSON escaped string as `value` in  `--filter-parameters name=value`
- `name` cannot be empty in `--filter-parameters name=value`
- Sample inputs for common data types:
    - String value: `--filter-parameters stringValue=\"This is some text\"`
    - Numeric value: `--filter-parameters numberValue=75`
    - Array value: `--filter-parameters arrayValue=[1,2,3]`
    - Boolean value: `--filter-parameters boolValue=true`
    - Object value: `--filter-parameters objectValue={\"Test\":{\"Level\":\"Debug\"}}`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
